### ISSUE TITLE
Layout/EmptyLineAfterGuardClause inserts empty line outside of block, breaking method chaining

### ISSUE THREAD

**Issue by afn** [rubocop/rubocop#9977]
The Layout/EmptyLineAfterGuardClause cop misbehaves in a situation such as this (very contrived) example:

```ruby
# frozen_string_literal: true

def weird_sum(my_array)
  my_array
    .map { |item| next unless item.positive?; item * 2 }
    .compact
    .sum
end
```

## Expected behavior

No empty line inserted, **or** an empty line is inserted inside the block.

## Actual behavior

An empty line is inserted before the `.compact` line, resulting in a syntax error.

```
$ rubocop --debug --auto-correct test.rb
For /home/afn/work: Default configuration from /home/afn/.rvm/gems/ruby-3.0.1/gems/rubocop-1.18.4/config/default.yml
Inspecting 1 file
Scanning /home/afn/work/test.rb
E

Offenses:

test.rb:5:19: C: [Corrected] Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
    .map { |item| next unless item.positive?; item * 2 }
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
test.rb:7:5: E: Lint/Syntax: unexpected token tDOT
(Using Ruby 3.0 parser; configure using TargetRubyVersion parameter, under AllCops)
    .compact
    ^

1 file inspected, 2 offenses detected, 1 offense corrected
Finished in 0.11396609200164676 seconds
```

Resulting file:

```ruby
# frozen_string_literal: true

def weird_sum(my_array)
  my_array
    .map { |item| next unless item.positive?; item * 2 }

    .compact
    .sum
end
```

## Steps to reproduce the problem

See above.

## RuboCop version

1.18.4 (using Parser 3.0.2.0, rubocop-ast 1.8.0, running on ruby 3.0.1 x86_64-linux)
